src/hotspot/share/utilities/vmError.cpp

```diff
@@ -1678,7 +1704,17 @@ void VMError::report_and_die(int id, const char* message, const char* detail_fmt
     OnError = nullptr;
   }
 
-  if (WINDOWS_ONLY(!UseOSErrorReporting) NOT_WINDOWS(true)) {
+  if (after_report_action == AfterReportAction::Return) {
+    // Caller requested reporting without termination.
+    return;
+  } else if (after_report_action == AfterReportAction::Die) {
```

Comment:
Nit: as there are only 2 values for ARA we don't really need an `else if` here.

Reply:
I didn't want to add an assert.  But see reply to your next comment, about the ShouldNotReachHere at the end.